In [1]:
!gdown 1g1nwk4k-h9FceEHKZc8ocfu_xp3xnZ8R

Downloading...
From: https://drive.google.com/uc?id=1g1nwk4k-h9FceEHKZc8ocfu_xp3xnZ8R
To: /content/hr.csv
100% 580k/580k [00:00<00:00, 86.6MB/s]


In [2]:
import pandas as pd
import seaborn as sns


In [ ]:
# Load dataset

df = pd.read_csv('/content/hr.csv')

In [ ]:
df.head()

satisfaction_level  last_evaluation  number_project  average_montly_hours  \
0                0.24             0.79               2                   175   
1                0.40             0.49               2                   155   
2                0.45             0.45               2                   154   
3                0.46             0.73               6                   256   
4                0.14             0.54               5                   275   

   time_spend_company  Work_accident left  promotion_last_5years   Department  \
0                   5              0    0                      0           IT   
1                   3              0    1                      0           hr   
2                   3              1    1                      0      support   
3                   4              0    0                      0  product_mng   
4                   4              1    0                      0        sales   

   salary   age  
0     low   NaN  
1  medium  27.0  
2  medium   NaN  
3    high   NaN  
4    high   NaN

In [ ]:
# We have description of numerical column of above dataset
df.describe()

satisfaction_level  last_evaluation  number_project  \
count        15004.000000     15004.000000    15004.000000   
mean             0.612871         0.716096        3.802919   
std              0.248628         0.171163        1.232550   
min              0.090000         0.360000        2.000000   
25%              0.440000         0.560000        3.000000   
50%              0.640000         0.720000        4.000000   
75%              0.820000         0.870000        5.000000   
max              1.000000         1.000000        7.000000   

       average_montly_hours  time_spend_company  Work_accident  \
count          15004.000000        15004.000000   15004.000000   
mean             201.044655            3.497934       0.144561   
std               49.937662            1.460012       0.351670   
min               96.000000            2.000000       0.000000   
25%              156.000000            3.000000       0.000000   
50%              200.000000            3.000000       0.000000   
75%              245.000000            4.000000       0.000000   
max              310.000000           10.000000       1.000000   

       promotion_last_5years          age  
count           15004.000000  3080.000000  
mean                0.021261    27.018506  
std                 0.144258     1.429511  
min                 0.000000    25.000000  
25%                 0.000000    26.000000  
50%                 0.000000    27.000000  
75%                 0.000000    28.000000  
max                 1.000000    29.000000

In [ ]:
# Checking number of rows and columns

rows, columns = df.shape
print(f'We have {rows} number of rows and {columns} number of columns in above dataset.')

We have 15004 number of rows and 11 number of columns in above dataset.


In [ ]:
# Information about above dataset

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15004 entries, 0 to 15003
Data columns (total 11 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   satisfaction_level     15004 non-null  float64
 1   last_evaluation        15004 non-null  float64
 2   number_project         15004 non-null  int64  
 3   average_montly_hours   15004 non-null  int64  
 4   time_spend_company     15004 non-null  int64  
 5   Work_accident          15004 non-null  int64  
 6   left                   15004 non-null  object 
 7   promotion_last_5years  15004 non-null  int64  
 8   Department             15004 non-null  object 
 9   salary                 14975 non-null  object 
 10  age                    3080 non-null   float64
dtypes: float64(3), int64(5), object(3)
memory usage: 1.3+ MB


info() method shows that we have 15004 rows but salary column has 14975 not-null rows i.e. this column has missing data and also age column has 3080 not-null data i.e. this column has also missing data. so we have to handle missing data.

In [ ]:
# Checking for null values

df.isna().sum()

satisfaction_level           0
last_evaluation              0
number_project               0
average_montly_hours         0
time_spend_company           0
Work_accident                0
left                         0
promotion_last_5years        0
Department                   0
salary                      29
age                      11924
dtype: int64

Above information shows that age column has 11924 missing data and salary has 29 missing data.

In [ ]:
df.isna().sum()/len(df)

satisfaction_level       0.000000
last_evaluation          0.000000
number_project           0.000000
average_montly_hours     0.000000
time_spend_company       0.000000
Work_accident            0.000000
left                     0.000000
promotion_last_5years    0.000000
Department               0.000000
salary                   0.001933
age                      0.794721
dtype: float64

In [ ]:
print(f'Percent of data missing in age column is {0.794721 * 100} %')
print(f'Percent of data missing in salary column is {0.001933 * 100} %')

Percent of data missing in age column is 79.4721 %
Percent of data missing in salary column is 0.1933 %


Above information shows that salary column has 0.19% missing data and age column has 79% missing data.

We are dropping age column because more than 70% data are missing.

In [ ]:
df.drop(columns = ['age'], inplace = True)

In [ ]:
# Find missing data for salary column
df.isna().sum()/len(df)

satisfaction_level       0.000000
last_evaluation          0.000000
number_project           0.000000
average_montly_hours     0.000000
time_spend_company       0.000000
Work_accident            0.000000
left                     0.000000
promotion_last_5years    0.000000
Department               0.000000
salary                   0.001933
dtype: float64

In [ ]:
print(f'Percent of data missing in salary column is {0.0019 * 100} %')

Percent of data missing in salary column is 0.19 %


Above information shows that salary has only 0.19% data missing so we drop missing rows.

In [ ]:
df.dropna(inplace = True)

In [ ]:
df.isna().sum()

satisfaction_level       0
last_evaluation          0
number_project           0
average_montly_hours     0
time_spend_company       0
Work_accident            0
left                     0
promotion_last_5years    0
Department               0
salary                   0
dtype: int64

# Encoding categorical column with numeric

In [ ]:
# Checking unique values in Department column

df.Department.unique()

array(['IT', 'hr', 'support', 'product_mng', 'sales', 'technical',
       'marketing', 'management', 'accounting', 'RandD'], dtype=object)

In [ ]:
# Checking unique values in salary column

df.salary.unique()

array(['low', 'medium', 'high'], dtype=object)

# Encoded categorical values using Label Encoder

In [ ]:
from sklearn.preprocessing import LabelEncoder

encoder = LabelEncoder()

df['Department'] = encoder.fit_transform(df['Department'])
df['salary'] = encoder.fit_transform(df['salary'])


In [ ]:
df.head()

satisfaction_level  last_evaluation  number_project  average_montly_hours  \
0                0.24             0.79               2                   175   
1                0.40             0.49               2                   155   
2                0.45             0.45               2                   154   
3                0.46             0.73               6                   256   
4                0.14             0.54               5                   275   

   time_spend_company  Work_accident left  promotion_last_5years  Department  \
0                   5              0    0                      0           0   
1                   3              0    1                      0           3   
2                   3              1    1                      0           8   
3                   4              0    0                      0           6   
4                   4              1    0                      0           7   

   salary  
0       1  
1       2  
2       2  
3       0  
4       0

Now at the end we handle the missing data.